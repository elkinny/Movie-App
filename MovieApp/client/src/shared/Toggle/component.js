import React from 'react';

import styles from './style.scss';

//@flow

type Props = {
  labels: ?Array<string>,
  onChange: ?(value: mixed) => mixed,
  text: ?string,
  name: ?string,
  defaultValue: ?boolean,
  values: ?Array<string>,
};

const Toggle = (props: Props) => {
  const { labels, text, name, onChange, values, defaultValue } = props;
  return (
    <div className={styles['toggle']}>
      <span className={styles['toggle__label']}> {text} </span>
      <div className={styles['toggle__body']}>
        <input
          id={'toggle-' + labels[0]}
          className={styles['toggle__item--left']}
          name={name}
          type="radio"
          value={values[0]}
          defaultChecked={!defaultValue}
          onChange={onChange}
        />
        <label htmlFor={'toggle-' + labels[0]} className={styles['btn']}>
          {labels[0]}
        </label>
        <input
          id={'toggle-' + labels[1]}
          className={styles['toggle__item--right']}
          name={name}
          value={values[1]}
          type="radio"
          defaultChecked={defaultValue}
          onChange={onChange}
        />
        <label htmlFor={'toggle-' + labels[1]} className={styles['btn']}>
          {labels[1]}
        </label>
      </div>
    </div>
  );
};

export default Toggle;
