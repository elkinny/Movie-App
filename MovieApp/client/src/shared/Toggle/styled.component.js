import styled from 'styled-components';

export const StyledToggle = styled.div`
  display: flex;
  align-items: center;

  @media screen and (max-width: 800px) {
    flex-direction: column;
    margin: 20px 0;
  }
`;

export const StyledToggleBtn = styled.label`
  border: 2px solid #7488ab;
  color: #7488ab;
  display: inline-block;
  padding: 10px 20px;
  position: relative;
  text-align: center;
  transition: background-color 600ms ease, color 600ms ease;
  user-select: none;
`;

export const StyledToggleLabel = styled.span`
  color: #7488ab;
  margin-right: 15px;
  margin-bottom: 0;

  @media screen and (max-width: 800px) {
    margin-right: 0;
    margin-bottom: 10px;
  }
`;

export const StyledToggleBody = styled.div`
  overflow: hidden;
`;

const leftAndRight = `
  display: none;

  & + ${StyledToggleBtn} {
    cursor: pointer;
    min-width: 60px;
  }

  & + ${StyledToggleBtn}:hover {
    background: none;
    color: #fff;
  }

  & + ${StyledToggleBtn}:after {
    background: #7488ab;
    content: '';
    height: 100%;
    position: absolute;
    top: 0;
    transition: left 200ms cubic-bezier(.77, 0, .175, 1);
    width: 100%;
    z-index: -1;
  }

  &:checked + ${StyledToggleBtn} {
    cursor: default;
    color: #17181c;
    transition: color 200ms;
  }

  &:checked + ${StyledToggleBtn}:after {
    left: 0;
  }
`;

export const StyledToggleItemLeft = styled.input`
  & + ${StyledToggleBtn} {
    border-right: 0;
  }

  & + ${StyledToggleBtn}:after {
    left: 100%;
  }

  ${leftAndRight}
`;

export const StyledToggleItemRight = styled.input`
  & + ${StyledToggleBtn} {
    margin-left: -5px;
  }

  & + ${StyledToggleBtn}:after {
    left: -100%;
  }

  ${leftAndRight}
`;
