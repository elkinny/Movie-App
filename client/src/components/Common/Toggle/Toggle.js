import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './toggle.scss';

export default class Toggle extends Component {
  state = {
    chkbox: true,
  };

  onChange = e => {
    this.setState({ chkbox: !this.state.chkbox });
    this.props.onChange(e);
  };

  render() {
    const { labels, text, type } = this.props;
    return (
      <div className={styles['toggle']}>
        <span className={styles['toggle__label']}> {text} </span>
        <div className={styles['toggle__body']}>
          <input
            id={'toggle-' + labels[0]}
            className={styles['toggle__item--left']}
            name={'toggled_' + type}
            type="radio"
            value={labels[0].toLowerCase()}
            defaultChecked={!this.state.chkbox}
            onChange={this.onChange}
          />
          <label htmlFor={'toggle-' + labels[0]} className={styles['btn']}>
            {labels[0]}
          </label>
          <input
            id={'toggle-' + labels[1]}
            className={styles['toggle__item--right']}
            name={'toggled_' + type}
            value={labels[1].toLowerCase()}
            type="radio"
            defaultChecked={this.state.chkbox}
            onChange={this.onChange}
          />
          <label htmlFor={'toggle-' + labels[1]} className={styles['btn']}>
            {labels[1]}
          </label>
        </div>
      </div>
    );
  }
}

Toggle.propTypes = {
  labels: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};
