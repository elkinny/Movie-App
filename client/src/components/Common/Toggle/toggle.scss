.toggle {
  .btn {
    border: 2px solid #7488ab;
    color: #7488ab;
    display: inline-block;
    padding: 10px 20px;
    position: relative;
    text-align: center;
    transition: background-color 600ms ease, color 600ms ease;
    user-select: none;
  }

  &__label {
    color: #7488ab;
    margin-right: 15px;
  }

  &__item {
    &--left + label {
      border-right: 0;

      &:after {
        left: 100%;
      }
    }

    &--right + label {
      margin-left: -5px;

      &:after {
        left: -100%;
      }
    }

    &--left,
    &--right {
      display: none;

      & + label {
        cursor: pointer;
        min-width: 60px;

        &:hover {
          background: none;
          color: #fff;
        }

        &:after {
          background: #7488ab;
          content: '';
          height: 100%;
          position: absolute;
          top: 0;
          transition: left 200ms cubic-bezier(.77, 0, .175, 1);
          width: 100%;
          z-index: -1;
        }
      }

      &:checked + label {
        cursor: default;
        color: #17181c;
        transition: color 200ms;

        &:after {
          left: 0;
        }
      }
    }
  }
}
