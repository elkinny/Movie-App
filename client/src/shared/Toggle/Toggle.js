import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './toggle.scss';

export default class Toggle extends Component {
  render() {
    const { labels, text, type, onChange, value } = this.props;
    return (
      <div className={styles['toggle']}>
        <span className={styles['toggle__label']}> {text} </span>
        <div className={styles['toggle__body']}>
          <input
            id={'toggle-' + labels[0]}
            className={styles['toggle__item--left']}
            name={type + 'Toggle'}
            type="radio"
            value={labels[0].toLowerCase()}
            defaultChecked={!value}
            onChange={onChange}
          />
          <label htmlFor={'toggle-' + labels[0]} className={styles['btn']}>
            {labels[0]}
          </label>
          <input
            id={'toggle-' + labels[1]}
            className={styles['toggle__item--right']}
            name={type + 'Toggle'}
            value={labels[1].toLowerCase()}
            type="radio"
            defaultChecked={value}
            onChange={onChange}
          />
          <label htmlFor={'toggle-' + labels[1]} className={styles['btn']}>
            {labels[1]}
          </label>
        </div>
      </div>
    );
  }
}

Toggle.propTypes = {
  labels: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  value: PropTypes.bool.isRequired,
};
