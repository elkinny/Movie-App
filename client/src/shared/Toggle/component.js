import React from 'react';
import PropTypes from 'prop-types';

import styles from './style.scss';

const Toggle = props => {
  const { labels, text, type, onChange, values, defaultValue } = props;
  return (
    <div className={styles['toggle']}>
      <span className={styles['toggle__label']}> {text} </span>
      <div className={styles['toggle__body']}>
        <input
          id={'toggle-' + labels[0]}
          className={styles['toggle__item--left']}
          name={type + 'Toggle'}
          type="radio"
          value={values[0]}
          defaultChecked={!defaultValue}
          onChange={onChange}
        />
        <label htmlFor={'toggle-' + labels[0]} className={styles['btn']}>
          {labels[0]}
        </label>
        <input
          id={'toggle-' + labels[1]}
          className={styles['toggle__item--right']}
          name={type + 'Toggle'}
          value={values[1]}
          type="radio"
          defaultChecked={defaultValue}
          onChange={onChange}
        />
        <label htmlFor={'toggle-' + labels[1]} className={styles['btn']}>
          {labels[1]}
        </label>
      </div>
    </div>
  );
};

Toggle.propTypes = {
  labels: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  defaultValue: PropTypes.bool.isRequired,
  values: PropTypes.array.isRequired,
};

export default Toggle;
