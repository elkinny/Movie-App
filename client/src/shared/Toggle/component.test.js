import Toggle from './Toggle';
import React from 'react';
import { mount } from './../../test/enzymeSetup.js';

describe('Toggle: ', () => {
  it('Toggle to match snapshot', () => {
    const props = {
      labels: ['Rating', 'Year'],
      type: 'sort',
      text: 'Sort by:',
      onChange: jest.fn(),
      value: true,
    };
    const component = mount(<Toggle {...props} />);
    expect(component).toMatchSnapshot();
  });
});
