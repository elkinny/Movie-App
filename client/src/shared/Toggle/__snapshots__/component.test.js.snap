// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle:  Toggle to match snapshot 1`] = `
<Toggle
  defaultValue={true}
  labels={
    Array [
      "Rate",
      "Date",
    ]
  }
  onChange={[MockFunction]}
  text="Sort by:"
  type="sort"
  values={
    Array [
      "vote_average",
      "release_date",
    ]
  }
>
  <div
    className="toggle"
  >
    <span
      className="toggle__label"
    >
       
      Sort by:
       
    </span>
    <div
      className="toggle__body"
    >
      <input
        className="toggle__item--left"
        defaultChecked={false}
        id="toggle-Rate"
        name="sortToggle"
        onChange={[MockFunction]}
        type="radio"
        value="vote_average"
      />
      <label
        className="btn"
        htmlFor="toggle-Rate"
      >
        Rate
      </label>
      <input
        className="toggle__item--right"
        defaultChecked={true}
        id="toggle-Date"
        name="sortToggle"
        onChange={[MockFunction]}
        type="radio"
        value="release_date"
      />
      <label
        className="btn"
        htmlFor="toggle-Date"
      >
        Date
      </label>
    </div>
  </div>
</Toggle>
`;
